jib-maven-plugin/src/main/java/com/google/cloud/tools/jib/maven/BuildImageMojo.java

```diff
@@ -83,19 +88,55 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
     ImageReference targetImage =
         PluginConfigurationProcessor.parseImageReference(getTargetImage(), "to");
-    Authorization toAuthorization =
-        ConfigurationPropertyValidator.getImageAuth(
+    CredentialRetrieverFactory credentialRetrieverFactory =
+        CredentialRetrieverFactory.forImage(targetImage, mavenJibLogger);
+    Credential toCredential =
+        ConfigurationPropertyValidator.getImageCredential(
             mavenJibLogger, "jib.to.auth.username", "jib.to.auth.password", getTargetImageAuth());
-    RegistryCredentials knownTargetRegistryCredentials =
-        toAuthorization != null
-            ? new RegistryCredentials("jib-maven-plugin <to><auth> configuration", toAuthorization)
-            : pluginConfigurationProcessor
-                .getMavenSettingsServerCredentials()
-                .retrieve(targetImage.getRegistry());
+    RegistryCredentials knownTargetRegistryCredentials = null;
+    CredentialRetriever knownCredentialRetriever = null;
+    if (toCredential == null) {
+      toCredential =
+          pluginConfigurationProcessor
+              .getMavenSettingsServerCredentials()
+              .retrieve(targetImage.getRegistry());
+      if (toCredential != null) {
+        knownTargetRegistryCredentials =
+            new RegistryCredentials(
+                MavenSettingsServerCredentials.CREDENTIAL_SOURCE,
+                Authorizations.withBasicCredentials(
+                    toCredential.getUsername(), toCredential.getPassword()));
+        knownCredentialRetriever =
+            credentialRetrieverFactory.known(
+                toCredential, MavenSettingsServerCredentials.CREDENTIAL_SOURCE);
+      }
+    } else {
+      knownTargetRegistryCredentials =
+          new RegistryCredentials(
+              "jib-maven-plugin <from><auth> configuration",
```

Comment:
"jib-maven-plugin \<to\>\<auth\> configuration"

Reply:
This is what happens when I copy-paste too much code 🤦‍♂️